lib/ansible/modules/system/ufw.py

```diff
@@ -336,14 +435,48 @@ def ufw_version():
             if (ufw_major == 0 and ufw_minor >= 35) or ufw_major > 0:
                 cmd.append([params['comment'], "comment '%s'" % params['comment']])
 
-            execute(cmd)
+            rules_dry = execute(cmd)
 
-    # Get the new state
-    (_, post_state, _) = module.run_command(ufw_bin + ' status verbose')
-    (_, post_rules, _) = module.run_command("grep '^### tuple' /lib/ufw/user.rules /lib/ufw/user6.rules /etc/ufw/user.rules /etc/ufw/user6.rules")
-    changed = (pre_state != post_state) or (pre_rules != post_rules)
+            if module.check_mode:
+                # if module.boolean(params['delete']):
+                # delete rules with --dry-run doesn't display "Rules updated" messages
+                # we compare only ## tupple
+                # also, ipv4 change
+
+                nb_skipping_line = len(filter_line_that_contains("Skipping", rules_dry))
+
+                if not (nb_skipping_line > 0 and nb_skipping_line == len(rules_dry.splitlines(True))):
+                    rules_dry = filter_line_that_not_start_with("### tuple", rules_dry)
 
-    return module.exit_json(changed=changed, commands=cmds, msg=post_state.rstrip())
+                    keys = ['from_ip', 'to_ip']
+                    for key in keys:
+                        if key not in params:
+                            key.remove(key)
```

Comment:
```suggestion
                            keys.remove(key)
```
maybe?

Instead of this `for` loop, why not have `keys = [key for key in keys if key in parms]` instead?

Also, both **will** be in `params`. You're more likely interested in the one which isn't `any`?

Reply:
yep I have removed all this part... it's not usefull in fact.